packages/remark-lint-final-definition/index.js

```diff
@@ -46,7 +46,7 @@ function finalDefinition(tree, file) {
     var line = start(node).line
 
     // Ignore generated nodes.
-    if (node.type === 'root' || generated(node)) {
+    if (node.type === 'root' || generated(node) || /^\s*<!--/.test(node.value)) {
```

Comment:
The `no-html` rule only works on `html` nodes for this check. Here we can have any node!

Could you perform the regex only if it is an `html` node? Something like: `(node.type === 'html' && /.../.test(node.value))`

Reply:
Oh and update the comment above!

Reply:
Sure thing